nsqd/stats.go

```diff
@@ -52,7 +53,7 @@ func NewChannelStats(c *Channel, clients []ClientStats) ChannelStats {
 		BackendDepth:  c.backend.Depth(),
 		InFlightCount: len(c.inFlightMessages),
 		DeferredCount: len(c.deferredMessages),
-		MessageCount:  c.messageCount,
+		MessageCount:  atomic.LoadUint64(&c.messageCount),
```

Comment:
If this is truly necessary (and I think you're right, it might be) then it's probably necessary for *all* of these fields that are modified atomically by `Topic` or `Channel`.

We should keep this change as a separate commit, too.

p.s. did you encounter some race condition?  what prompted the change?